applications/nrf5340_audio/src/modules/audio_usb.c

```diff
@@ -7,26 +7,44 @@
 #include "audio_usb.h"
 
 #include <zephyr/kernel.h>
-#include <zephyr/usb/usb_device.h>
-#include <zephyr/usb/class/usb_audio.h>
+#include <zephyr/usb/usbd.h>
+#include <zephyr/usb/class/usbd_uac2.h>
 #include <zephyr/bluetooth/audio/audio.h>
+#include <zephyr/net_buf.h>
 #include <audio_defines.h>
 
 #include "macros_common.h"
 #include "device_location.h"
+#include <pcm_stream_channel_modifier.h>
 
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(audio_usb, CONFIG_MODULE_AUDIO_USB_LOG_LEVEL);
 
+#define HEADSET_OUT_TERMINAL_ID	   UAC2_ENTITY_ID(DT_NODELABEL(out_terminal))
+#define HEADPHONES_OUT_TERMINAL_ID UAC2_ENTITY_ID(DT_NODELABEL(hp_out_terminal))
+#define HEADSET_IN_TERMINAL_ID	   UAC2_ENTITY_ID(DT_NODELABEL(in_terminal))
+
+/* Absolute minimum is 2 TX buffers but add 2 additional buffers to prevent out of memory
+ * errors when USB host decides to perform rapid terminal enable/disable cycles.
+ */
+#define USB_BLOCKS 4
+
+/* See udc_buf.h for more information about UDC_BUF_GRANULARITY and UDC_BUF_ALIGN */
+K_MEM_SLAB_DEFINE_STATIC(usb_tx_slab, ROUND_UP(USB_BLOCK_SIZE_MULTI_CHAN, UDC_BUF_GRANULARITY),
+			 USB_BLOCKS, UDC_BUF_ALIGN);
+K_MEM_SLAB_DEFINE_STATIC(usb_rx_slab, ROUND_UP(USB_BLOCK_SIZE_MULTI_CHAN, UDC_BUF_GRANULARITY),
+			 USB_BLOCKS, UDC_BUF_ALIGN);
+
 static struct k_msgq *audio_q_tx;
 static struct k_msgq *audio_q_rx;
 
 NET_BUF_POOL_FIXED_DEFINE(pool_in, CONFIG_FIFO_FRAME_SPLIT_NUM, USB_BLOCK_SIZE_MULTI_CHAN,
 			  sizeof(struct audio_metadata), NULL);
 
 static uint32_t rx_num_overruns;
-static bool rx_first_data;
-static bool tx_first_data;
+static bool headset_out_terminal_enabled;
```

Comment:
Nit: IMO they should all be named `terminal_xxx ` as that is the most generic part of the name.